src/runtime_src/core/common/info_aie.cpp

```diff
@@ -327,6 +323,72 @@ populate_aie_shim(const xrt_core::device *device, const std::string& desc)
   return pt;
 }
 
+// Populate the AIE Mem tile information from the input XRT device
+boost::property_tree::ptree
+populate_aie_mem(const xrt_core::device *device, const std::string& desc)
+{
+  boost::property_tree::ptree pt;
+  pt.put("description", desc);
+  boost::property_tree::ptree pt_mem;
+
+  try {
+    // Read AIE Mem information of the device
+    std::string aie_data = xrt_core::device_query<qr::aie_mem_info>(device);
+    std::stringstream ss(aie_data);
+    boost::property_tree::read_json(ss, pt_mem);
+  }
+  catch (const xrt_core::query::no_such_key&) {
+    // Memory tile is not present in all aie devices
+    pt.put("error_msg", "AIE mem tile information is not present");
+    return pt;
+  }
+  catch (const std::exception& ex) {
+    pt.put("error_msg", ex.what());
+    return pt;
+  }
+
+  try {
+    boost::property_tree::ptree tile_array;
+
+    // Populate the mem tile information such as dma, lock, error, events
+    // for each tiles.
+    for (const auto& am: pt_mem.get_child("aie_mem")) {
+      const boost::property_tree::ptree& imem = am.second;
+      boost::property_tree::ptree omem;
+      int col = imem.get<uint32_t>("col");
+      int row = imem.get<uint32_t>("row");
+
+      omem.put("column", col);
+      omem.put("row", row);
+
+      // Populate DMA information
+      if (imem.find("dma") != imem.not_found())
+        populate_aie_dma(imem, omem);
+
+      // Populate ERROR information
+      if (imem.find("errors") != imem.not_found())
+        populate_aie_errors(imem, omem);
+
+      // Populate LOCK information
+      if (imem.find("lock") != imem.not_found())
+        addnodelist("lock", "locks", imem, omem);
+
+      // Populate EVENT information
+      if (imem.find("event") != imem.not_found())
+        addnodelist("event", "events", imem, omem);
+
+      tile_array.push_back(std::make_pair("tile" + std::to_string(col), omem));
```

Comment:
Remove all the `make_pair` of the changes
```suggestion
      tile_array.push_back({ "tile" + std::to_string(col), omem });
```